tools/ci-operator-prowgen/README.md

```diff
@@ -0,0 +1,140 @@
+# Prow job generator for ci-operator
+
+The purpose of this tool is to reduce an amount of boilerplate that component
+owners need to write when they use
+[ci-operator](https://github.com/openshift/ci-operator) to set up CI for their
+component. The generator is able to entirely generate the necessary Prow job
+configuration from the ci-operator configuration file.
+
+## Use
+
+To use the generator, you need to build it:
+
+```
+$ go build
+```
+
+### Full-repository
+
+The generator can use the naming convention and directory structure of this
+repository. Using this mode, all you need to do is to place your `ci-operator`
+configuration file to the correct place in
+[ci-operator/config](../../ci-operator/config) directory and then run the generator
+with `--full-repo` parameter:
+
+```
+$ ./ci-operator-prowgen --full-repo
```

Comment:
When we shove this into a container to run it, we add a dependency on `git`. Not sure it's super useful in addition to the explcit dir-as-flags approach if we can just add a script to this repo to run the generator with the appropriate dirs, but I guess if we ensure we thread through the dependencies as necessary it should be fine either way

Reply:
I think this is a question of main use case - right now, do we intend the generator to be used by humans or do we want it to be shoved into some automated script? If the former then I would really like to keep it (not fond of wrapping stuff in scripts so it's usable).